CI tests for metalk8s_monitoring and metalk8s_kubernetes_cronjob salt modules

aprucolimartins opened the pull request:
**Component**:

<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: 

**Summary**:

**Acceptance criteria**: 


---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #ISSUE_NUMBER

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

aprucolimartins committed:
tests: make "Given the Prometheus API is available" shared

Move the `check_prometheus_api()` function from `test_monitoring.py`
to `conftest.py` in order to make it available for all test modules

Co-authored-by: Alex Rodriguez <alex.rodriguez@scality.com>

aprucolimartins committed:
tests: add "Given the Salt Master is available" shared

Add the `check_salt_master()` function to the `conftest.py` file
in order to make it available for all tests

Co-authored-by: Alex Rodriguez <alex.rodriguez@scality.com>

aprucolimartins committed:
tests: add CI tests for the `metalk8s_monitoring` salt module

Add CI test for the `metalk8s_monitoring` salt module in order
to check if module is capable to:

    1 - list silences
    2 - list alerts
    3 - create silences
    4 - delete silences
    5 - verify that silences work

Co-authored-by: Alex Rodriguez <alex.rodriguez@scality.com>
Co-authored-by: Teddy Andrieux <teddy.andrieux@scality.com>

bert-e commented:
# Hello aprucolimartins,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/metalk8s/doc/user).





<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Request integration branches



Waiting for integration branch creation to be requested by the user.

To request integration branches, please comment on this pull request with the following command:

```
/create_integration_branches
```

Alternatively, the `/approve` and `/create_pull_requests` commands will automatically
create the integration branches.

aprucolimartins committed:
test: make `negation()` function shared

Move the `negation()` function from `test_salt_api.py` to
`common.py` to make it available to all tests.

Co-authored-by: Alex Rodriguez <alex.rodriguez@scality.com>

ezekiel-alexrod committed:
tests: add CI tests for the `metalk8s_kubernetes_cronjob` salt module

    Add CI test for the `metalk8s_kubernetes_cronjob` salt module in order
    to check if module is capable to:
        1 - We can list the CronJobs be different criteria
        2 - We can suspend, mark, and activate CronJobs
        3 - We can stop all Jobs fired by a CronJob

Co-authored-by: Antonisio (SpaceDog) PRUCOLI MARTINS <antonisio.prucoli.martins@scality.com>
Co-authored-by: Teddy Andrieux <teddy.andrieux@scality.com>

TeddyAndrieux commented on tests/post/features/salt_metalk8s_kubernetes_cronjob.feature:
```diff
@@ -0,0 +1,35 @@
+        When we activate the 'test-cronjob' CronJob
+        And we list the 'active' CronJobs
+        Then the 'test-cronjob' CronJob should be in the list of 'active' CronJobs
+        And the 'test-cronjob' CronJob should not be marked with 'test-suspend-mark'
```
It's the same comment as I did on monitoring, each scenario should work independently

TeddyAndrieux replied:
To me at the point it shouldn't be marked at all no ?

TeddyAndrieux commented on tests/post/features/salt_metalk8s_kubernetes_cronjob.feature:
```diff
@@ -0,0 +1,35 @@
+        Then the 'test-cronjob' CronJob should be in the list of 'active' CronJobs
+        And the 'test-cronjob' CronJob should not be marked with 'test-suspend-mark'
+
+    Scenario: Get a CronJob's jobs
+        When we wait for '60' seconds
+        And we get the 'test-cronjob' jobs
+        Then the 'test-cronjob' jobs should not be empty
```
We just test CronJob here, I don't feel this Scenario is really usefull

aprucolimartins replied:
The requirement to the feature is to be capable to stop all jobs associated to the CronJobs we stopped. So, in our opinion, this check is mandatory.

TeddyAndrieux replied:
Agree but it should be part of the next scenario as "Given the 'test-cronjob' has some jobs" 

Again scenarios are supposed to work independently

TeddyAndrieux commented on tests/post/features/salt_metalk8s_kubernetes_cronjob.feature:
```diff
@@ -0,0 +1,35 @@
+        Then the 'test-cronjob' CronJob should be in the list of 'suspended' CronJobs
+        And the 'test-cronjob' CronJob should be marked with 'test-suspend-mark'
+        When we get the 'test-cronjob' jobs
+        Then the 'test-cronjob' jobs should be empty
```
It's a bit weird syntax here because the last When do not really perform any action 

I would do it as

```suggestion
        When we stop the 'test-cronjob' CronJob and delete all its Jobs marked with 'test-suspend-mark'
        And we list the 'suspended' CronJobs
        And we get the 'test-cronjob' jobs
        Then the 'test-cronjob' CronJob should be in the list of 'suspended' CronJobs
        And the 'test-cronjob' CronJob should be marked with 'test-suspend-mark'
        And the 'test-cronjob' jobs should be empty
```

TeddyAndrieux commented on tests/post/features/salt_metalk8s_monitoring.feature:
```diff
@@ -0,0 +1,28 @@
+        When we create a silence for the 'Watchdog' alert
+        And we list the 'active' silences
+        Then we should find the silence for the 'Watchdog' alert in the 'active' silences list
+        When we wait for '300' seconds
```
Can we do the retry check logic right now

aprucolimartins replied:
Done.

TeddyAndrieux commented on tests/post/steps/common.py:
```diff
@@ -0,0 +1,13 @@
+    elif value in [" not", "not"]:
+        return True
+    else:
+        raise ValueError("Cannot parse '{}' as an optional negation".format(value))
```
Could be moved to "tests/utils.py" directly

TeddyAndrieux commented on tests/post/steps/files/cronjob.yaml:
```diff
@@ -0,0 +1,16 @@
+        spec:
+          containers:
+            - name: test-container
+              image: busybox
```
Let's not sure busybox image but metalk8s utils one (like we do for logger-pod.yaml) 

(Otherwise it means we need internet to run tests)

Check `set_up_logger_pod` in `tests/post/steps/test_logging.py`

TeddyAndrieux commented on tests/post/steps/test_salt_metalk8s_kubernetes_cronjob.py:
```diff
@@ -0,0 +1,286 @@
+        if exc.status != 404:
+            raise exc
+
+    cj_k8s_client.create(body=cj_manifest, namespace="default")
```
To me we want the Given to create on CronJob per scenario and automatically delete it at the end, so I suggest really do to the same logic as for `set_up_logger_pod`

Because here it means at the end of all tests you will let the CronJob which is not what we want, we want that at the end of the tests the platform is exactly the same as before

aprucolimartins replied:
Done.

TeddyAndrieux commented on tests/post/steps/test_salt_metalk8s_kubernetes_cronjob.py:
```diff
@@ -0,0 +1,286 @@
+    # True if suspended, false if active
+    command.append(f"suspended={state == 'suspended'}")
+    command.append("--out json")
+    command.append("--log-level quiet")
```
nit: The order is not really important you could move those args directly in `command` definition right after `salt-run`

TeddyAndrieux commented on tests/post/steps/test_salt_metalk8s_kubernetes_cronjob.py:
```diff
@@ -0,0 +1,286 @@
+
+@when(
+    parsers.parse(
+        "we suspend the '{cronjob_name}' CronJob " "marking it with '{suspend_mark}'",
```
```suggestion
        "we suspend the '{cronjob_name}' CronJob marking it with '{suspend_mark}'",
```

TeddyAndrieux commented on tests/post/steps/test_salt_metalk8s_kubernetes_cronjob.py:
```diff
@@ -0,0 +1,286 @@
+        command,
+        ssh_config,
+    )
+    assert json.loads(result.stdout) == True, "Failed to suspend the CronJob"
```
nit

```suggestion
    assert json.loads(result.stdout) is True, "Failed to suspend the CronJob"
```

TeddyAndrieux commented on tests/post/steps/test_salt_metalk8s_kubernetes_cronjob.py:
```diff
@@ -0,0 +1,286 @@
+    assert json.loads(result.stdout) == True, "Failed to activate the CronJob"
+
+
+@when(
+    parsers.parse("we wait for '{seconds}' seconds"),
+)
+def wait_seconds(seconds):
+    time.sleep(int(seconds))
```
It does not seems used here

TeddyAndrieux commented on tests/post/steps/test_salt_metalk8s_kubernetes_cronjob.py:
```diff
@@ -0,0 +1,286 @@
+        .get("annotations", {})
+        .get("metalk8s.scality.com/suspend_mark")
+    )
+    if negated:
+        assert (
+            not suspend_annotation or suspend_annotation != suspend_mark
+        ), "CronJob has a suspend_mark"
```
To me it should either be "marked with ...." or "not marked" but there is no case where we would use "not marked with ..." no ?

TeddyAndrieux commented on tests/post/steps/test_salt_metalk8s_monitoring.py:
```diff
@@ -0,0 +1,206 @@
+        raise ValueError(f"Invalid silence state: {silence_state}")
+    if silence_state != "all":
+        command.append(f"state={silence_state}")
+    command.append("--out json")
+    command.append("--log-level quiet")
```
ditto: nit: Those 2 could be move the `command` declaration

aprucolimartins committed:
salt: fix use alertmanager API v2 on metalk8s_monitoring salt module

    1 - fix the response content parsing
    2 - fix the unit tests

Co-authored-by: Alex Rodriguez <alex.rodriguez@scality.com>

aprucolimartins committed:
salt: standardization names in `metalk8s_kubernetes_cronjob` salt module

	1 - `list_cronjobs` function become `get_cronjobs`
        2 - `stop_jobs` function become `suspend_cronjob_and_delete_jobs`
        3 - `cronjob_name` parameters become `name`

Co-authored-by: Alex Rodriguez <alex.rodriguez@scality.com>

TeddyAndrieux commented on tests/post/steps/test_salt_metalk8s_kubernetes_cronjob.py:
```diff
@@ -0,0 +1,287 @@
+        "image"
+    ] = utils_image
+    cj_k8s_client = k8s_client.resources.get(api_version="batch/v1", kind="CronJob")
+    try:
+        cj_k8s_client.delete(name=cronjob_name, namespace="default")
+    except ApiException as exc:
+        if exc.status != 404:
+            raise exc
```
To me this is not needed, each given is responsible of the deletion

aprucolimartins replied:
Removed.

TeddyAndrieux commented on tests/post/steps/test_salt_metalk8s_kubernetes_cronjob.py:
```diff
@@ -0,0 +1,287 @@
+            raise exc
+    cj_k8s_client.create(body=cj_manifest, namespace="default")
+
+    yield True
```
nit: Just `yield` is enough here

```suggestion
    yield
```

aprucolimartins replied:
Changed.

TeddyAndrieux commented on tests/post/steps/test_salt_metalk8s_kubernetes_cronjob.py:
```diff
@@ -0,0 +1,287 @@
+    parsers.parse("we list the '{state}' CronJobs"),
+)
+def call_metalk8s_kubernetes_cronjob_get_cronjobs(host, ssh_config, context, state):
+    command = [
+        "salt-run",
+        "salt.cmd",
+        "metalk8s_kubernetes_cronjob.get_cronjobs",
+        "namespace=default",
+        "--out json",
+        "--log-level quiet",
+    ]
+    if state not in ["active", "suspended"]:
+        raise ValueError(f"Invalid silence state: {state}")
+    # True if suspended, false if active
+    command.append(f"suspended={state == 'suspended'}")
```
nit

```suggestion
    if state not in ["active", "suspended"]:
        raise ValueError(f"Invalid silence state: {state}")

    command = [
        "salt-run",
        "salt.cmd",
        "metalk8s_kubernetes_cronjob.get_cronjobs",
        "namespace=default",
        # True if suspended, false if active
        f"suspended={state == 'suspended'}"),
        "--out json",
        "--log-level quiet",
    ]
```

aprucolimartins replied:
Changed.

TeddyAndrieux commented on tests/post/steps/test_salt_metalk8s_kubernetes_cronjob.py:
```diff
@@ -0,0 +1,287 @@
+        command,
+        ssh_config,
+    )
+    context[f"deleted_jobs"] = json.loads(result.stdout)
```
```suggestion
    context["deleted_jobs"] = json.loads(result.stdout)
```

Edit: This one does not seems used btw

aprucolimartins replied:
Fixed.

TeddyAndrieux commented on tests/post/steps/test_salt_metalk8s_monitoring.py:
```diff
@@ -0,0 +1,219 @@
+    parsers.parse("we list the '{alert_state}' alerts"),
+)
+def call_metalk8s_monitoring_get_alerts(host, ssh_config, context, alert_state):
+    command = [
+        "salt-run",
+        "salt.cmd",
+        "metalk8s_monitoring.get_alerts",
+    ]
+    if alert_state not in ["all", "active", "suppressed"]:
+        raise ValueError(f"Invalid alert state: {alert_state}")
+    if alert_state != "all":
+        command.append(f"state={alert_state}")
+    command.append("--out json")
+    command.append("--log-level quiet")
```
nit
```suggestion
    command = [
        "salt-run",
        "salt.cmd",
        "metalk8s_monitoring.get_alerts",
        "--out json",
        "--log-level quiet",
    ]
    if alert_state not in ["all", "active", "suppressed"]:
        raise ValueError(f"Invalid alert state: {alert_state}")
    if alert_state != "all":
        command.append(f"state={alert_state}")
```

aprucolimartins replied:
Changed.

TeddyAndrieux commented on tests/post/steps/test_salt_metalk8s_monitoring.py:
```diff
@@ -0,0 +1,219 @@
+    ),
+)
+def poll_alerts(host, ssh_config, context, alert_state, alert_name, timeout):
+    start = time.time()
+    while time.time() - start < int(timeout):
+        call_metalk8s_monitoring_get_alerts(host, ssh_config, context, alert_state)
+        result = any(
+            alert["labels"]["alertname"] == alert_name
+            for alert in context[f"alerts_{alert_state}"]
+        )
+        if result:
+            return
+        time.sleep(10)
+    assert False, f"{alert_name} alert not found in {alert_state} alerts list"
```
You should use `utils.retry` it's what we used everywhere to do such retry

aprucolimartins replied:
Done.

TeddyAndrieux commented on tests/post/steps/test_salt_metalk8s_kubernetes_cronjob.py:
```diff
@@ -0,0 +1,287 @@
+    ),
+)
+def poll_get_jobs(host, ssh_config, context, cronjob_name, timeout):
+    start_time = time.time()
+    while time.time() - start_time < timeout:
+        call_metalk8s_kubernetes_cronjob_get_jobs(
+            host, ssh_config, context, cronjob_name
+        )
+        if context[f"{cronjob_name}_jobs"]:
+            return
+        time.sleep(10)
+    assert False, f"CronJob {cronjob_name} has no jobs after {timeout} seconds"
```
ditto you should use `utils.retry`

aprucolimartins replied:
Done.

aprucolimartins commented:
/create_integration_branches

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/CI_tests_for_metalk8s_monitoring_and_metalk8s_kubernetes_cronjob_modules` into
`development/126.0`
* [w/127.0/improvement/CI_tests_for_metalk8s_monitoring_and_metalk8s_kubernetes_cronjob_modules](https://github.com/scality/metalk8s/compare/development/127.0...w/127.0/improvement/CI_tests_for_metalk8s_monitoring_and_metalk8s_kubernetes_cronjob_modules) will be merged into `development/127.0` 



The following branches will **NOT** be impacted:

* `development/123.0`
* `development/124.0`
* `development/124.1`
* `development/125.0`
* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.11`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @teddyandrieux

* @chengyanjin

* @aprucolimartins

* @m4nch0t

* @thomasdanan

* @xaf-scality

* @eg-ayoub

* @jbwatenbergscality

* @ezekiel-alexrod










*The following options are set:* **create_integration_branches**

bert-e commented:
# History mismatch



Merge commit #4175a56717573be5a741f48891a4589d04c7110f on the integration branch
`w/127.0/improvement/CI_tests_for_metalk8s_monitoring_and_metalk8s_kubernetes_cronjob_modules` is merging a branch which is neither the current
branch `improvement/CI_tests_for_metalk8s_monitoring_and_metalk8s_kubernetes_cronjob_modules` nor the development branch
`development/127.0`.

It is likely due to a rebase of the branch `improvement/CI_tests_for_metalk8s_monitoring_and_metalk8s_kubernetes_cronjob_modules` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_integration_branches**

aprucolimartins commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **create_integration_branches**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/CI_tests_for_metalk8s_monitoring_and_metalk8s_kubernetes_cronjob_modules` into
`development/126.0`
* [w/127.0/improvement/CI_tests_for_metalk8s_monitoring_and_metalk8s_kubernetes_cronjob_modules](https://github.com/scality/metalk8s/compare/development/127.0...w/127.0/improvement/CI_tests_for_metalk8s_monitoring_and_metalk8s_kubernetes_cronjob_modules) will be merged into `development/127.0` 



The following branches will **NOT** be impacted:

* `development/123.0`
* `development/124.0`
* `development/124.1`
* `development/125.0`
* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.11`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @teddyandrieux

* @chengyanjin

* @aprucolimartins

* @m4nch0t

* @thomasdanan

* @xaf-scality

* @eg-ayoub

* @jbwatenbergscality

* @ezekiel-alexrod










*The following options are set:* **create_integration_branches**

bert-e commented:
# History mismatch



Merge commit #1893ee92f19b0456d77a107558b6115a161e686e on the integration branch
`w/127.0/improvement/CI_tests_for_metalk8s_monitoring_and_metalk8s_kubernetes_cronjob_modules` is merging a branch which is neither the current
branch `improvement/CI_tests_for_metalk8s_monitoring_and_metalk8s_kubernetes_cronjob_modules` nor the development branch
`development/127.0`.

It is likely due to a rebase of the branch `improvement/CI_tests_for_metalk8s_monitoring_and_metalk8s_kubernetes_cronjob_modules` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_integration_branches**

aprucolimartins commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **create_integration_branches**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/CI_tests_for_metalk8s_monitoring_and_metalk8s_kubernetes_cronjob_modules` into
`development/126.0`
* [w/127.0/improvement/CI_tests_for_metalk8s_monitoring_and_metalk8s_kubernetes_cronjob_modules](https://github.com/scality/metalk8s/compare/development/127.0...w/127.0/improvement/CI_tests_for_metalk8s_monitoring_and_metalk8s_kubernetes_cronjob_modules) will be merged into `development/127.0` 



The following branches will **NOT** be impacted:

* `development/123.0`
* `development/124.0`
* `development/124.1`
* `development/125.0`
* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.11`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @teddyandrieux

* @chengyanjin

* @aprucolimartins

* @m4nch0t

* @thomasdanan

* @xaf-scality

* @eg-ayoub

* @jbwatenbergscality

* @ezekiel-alexrod










*The following options are set:* **create_integration_branches**

aprucolimartins commented:
/approve

aprucolimartins commented:
/wait

bert-e commented:
# History mismatch



Merge commit #939814bfc973268c88d9bb2702add142bcc70be6 on the integration branch
`w/127.0/improvement/CI_tests_for_metalk8s_monitoring_and_metalk8s_kubernetes_cronjob_modules` is merging a branch which is neither the current
branch `improvement/CI_tests_for_metalk8s_monitoring_and_metalk8s_kubernetes_cronjob_modules` nor the development branch
`development/127.0`.

It is likely due to a rebase of the branch `improvement/CI_tests_for_metalk8s_monitoring_and_metalk8s_kubernetes_cronjob_modules` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve, create_integration_branches**

aprucolimartins commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve, create_integration_branches**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/CI_tests_for_metalk8s_monitoring_and_metalk8s_kubernetes_cronjob_modules` into
`development/126.0`
* [w/127.0/improvement/CI_tests_for_metalk8s_monitoring_and_metalk8s_kubernetes_cronjob_modules](https://github.com/scality/metalk8s/compare/development/127.0...w/127.0/improvement/CI_tests_for_metalk8s_monitoring_and_metalk8s_kubernetes_cronjob_modules) will be merged into `development/127.0` 



The following branches will **NOT** be impacted:

* `development/123.0`
* `development/124.0`
* `development/124.1`
* `development/125.0`
* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.11`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve, create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @teddyandrieux

* @chengyanjin

* @aprucolimartins

* @m4nch0t

* @thomasdanan

* @xaf-scality

* @eg-ayoub

* @jbwatenbergscality

* @ezekiel-alexrod










*The following options are set:* **approve, create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @teddyandrieux

* @chengyanjin

* @aprucolimartins

* @m4nch0t

* @thomasdanan

* @xaf-scality

* @eg-ayoub

* @jbwatenbergscality

* @ezekiel-alexrod










*The following options are set:* **create_integration_branches**

aprucolimartins commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @teddyandrieux

* @chengyanjin

* @aprucolimartins

* @m4nch0t

* @thomasdanan

* @xaf-scality

* @eg-ayoub

* @jbwatenbergscality

* @ezekiel-alexrod










*The following options are set:* **approve, create_integration_branches**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/126.0`

* :heavy_check_mark: `development/127.0`



The following branches will **NOT** be impacted:

* `development/123.0`
* `development/124.0`
* `development/124.1`
* `development/125.0`
* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.11`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve, create_integration_branches**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/126.0`

* :heavy_check_mark: `development/127.0`



The following branches have **NOT** changed:

* `development/123.0`
* `development/124.0`
* `development/124.1`
* `development/125.0`
* `development/2.0`
* `development/2.1`
* `development/2.10`
* `development/2.11`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



Please check the status of the associated issue None.

Goodbye aprucolimartins.
